web_ui/src/shared/components/media-items-list/media-items-list.component.tsx

```diff
@@ -1,52 +1,84 @@
 // Copyright (C) 2022-2025 Intel Corporation
 // LIMITED EDGE SOFTWARE DISTRIBUTION LICENSE
 
-import { forwardRef, ReactNode, useMemo } from 'react';
+import { ReactNode, useRef } from 'react';
 
-import { View } from '@adobe/react-spectrum';
-import { StyleProps } from '@react-types/shared';
-import { VirtuosoGrid, VirtuosoGridHandle } from 'react-virtuoso';
+import { DimensionValue, View } from '@adobe/react-spectrum';
+import { useLoadMore } from '@react-aria/utils';
+import { Responsive } from '@react-types/shared';
+import {
+    ListBox as AriaComponentsListBox,
+    GridLayout,
+    ListBoxItem,
+    ListLayout,
+    Size,
+    Virtualizer,
+} from 'react-aria-components';
 
 import { VIEW_MODE_SETTINGS, ViewModes } from '../media-view-modes/utils';
 
 import classes from './media-items-list.module.scss';
 
-interface MediaItemsListProps extends StyleProps {
+type ViewModeSettings = Record<ViewModes, { size?: number; gap: number; maxColumns?: number; minItemSize?: number }>;
+interface MediaItemsListProps<T> {
     id?: string;
     ariaLabel?: string;
     viewMode: ViewModes;
-    totalCount: number;
-    endReached: () => void;
-    itemContent: (index: number) => ReactNode;
+    mediaItems: T[];
+    height?: Responsive<DimensionValue>;
+    viewModeSettings?: ViewModeSettings;
+    endReached?: () => void;
+    itemContent: (item: T) => ReactNode;
+    idFormatter: (item: T) => string;
 }
 
-// Each thumbnail has approximately 150 pixel height (this differs per resolution
-// as well as the large/medium/small thumbnails).
-// We want to overscan a few (4) rows of thumbnails so that when the user scrolls
-// they won't have to wait for loading the images of the next rows
-const OVERSCAN = 150 * 4;
-
-export const MediaItemsList = forwardRef<VirtuosoGridHandle | null, MediaItemsListProps>(
-    ({ id, viewMode, totalCount, itemContent, endReached, ariaLabel, ...styleProps }, ref): JSX.Element => {
-        const style = useMemo(() => {
-            return { ...VIEW_MODE_SETTINGS[viewMode] };
-        }, [viewMode]);
-
-        return (
-            <View id={id} height={'100%'} width={'100%'} {...styleProps}>
-                <VirtuosoGrid
-                    aria-label={ariaLabel}
+export const MediaItemsList = <T,>({
+    id,
+    height,
+    viewMode,
+    ariaLabel,
+    mediaItems,
+    viewModeSettings = VIEW_MODE_SETTINGS,
+    itemContent,
+    endReached,
+    idFormatter,
+}: MediaItemsListProps<T>): JSX.Element => {
+    const config = viewModeSettings[viewMode];
+    const isDetails = viewMode === ViewModes.DETAILS;
+    const formattedMediaItems = mediaItems?.map((item) => ({ id: idFormatter(item), ...item })) ?? [];
```

Comment:
Could we change the interface of `MediaItemsList` so that instead of `T[]` it takes `Array<T & { id: string }>` (or do something like `<T extends { id: string }>`?

That way we could avoid having to add the `idFormatter` as well as avoid creating a new `formattedMediaItems` array (which could become big if we are dealing with large datasets).

Reply:
That's a good point, but from what I’ve seen, the current dataset doesn’t use ids, only other identifiers. So I figured centralizing it here would require fewer changes across files. If we go with the id approach, we’d need to format them externally anyway.

Reply:
Hmm what about in that case only using `idFormatter` where it is needed?

The thing I'd like to avoid is copying the `mediaItems` because those might be a lot bigger than the items that are being rendered.

Reply:
My last commit removed the additional formatting, This is the new approach we're using: https://github.com/open-edge-platform/geti/pull/207/files#diff-15aa145ff96eb0b7610f843256cd08e50aecf0b4a669644f2f68a148c8f57a31R79